src/python/pants/backend/python/macros/common_requirements_rule.py

```diff
@@ -128,7 +128,9 @@ def generate_tgt(
         normalized_proj_name = canonicalize_project_name(project_name)
         tgt_overrides = overrides.pop(normalized_proj_name, {})
         if Dependencies.alias in tgt_overrides:
-            tgt_overrides[Dependencies.alias] = list(tgt_overrides[Dependencies.alias]) + req_deps
+            tgt_overrides = tgt_overrides | {
```

Comment:
Are you avoiding changing the original tgt_overrides var?

Reply:
Yea.. it's because a mapping type was changed from mutable to immutable.
https://github.com/pantsbuild/pants/pull/21027/files#diff-9d70c6a6600a093491bdc5af25e125f07e92e700971dd544ba51ff8c3e6b186cR1184

Reply:
Ah. Makes sense.